workspaces/rbac/plugins/rbac/src/components/EditRole.tsx

```diff
@@ -22,26 +22,44 @@ import EditIcon from '@mui/icons-material/Edit';
 import IconButton from '@mui/material/IconButton';
 import Tooltip from '@mui/material/Tooltip';
 import Typography from '@mui/material/Typography';
+import { usePermission } from '@backstage/plugin-permission-react';
+import { policyEntityUpdatePermission } from '@backstage-community/plugin-rbac-common';
 
 type EditRoleProps = {
   roleName: string;
-  disable: boolean;
+  canEdit: boolean;
+  dataTestId?: string;
   tooltip?: string;
-  dataTestId: string;
   to?: string;
 };
 
 const EditRole = ({
   roleName,
-  tooltip,
-  disable,
+  canEdit,
   dataTestId,
+  tooltip,
   to,
 }: EditRoleProps) => {
   const { name, namespace, kind } = parseEntityRef(roleName);
+
+  const editPermissionResult = usePermission({
+    permission: policyEntityUpdatePermission,
+    resourceRef: roleName,
+  });
+
+  const dataTestIdText = !(editPermissionResult.allowed && canEdit)
+    ? `disable-update-role-${roleName}`
+    : `update-role-${roleName}`;
+
+  const disable = !(editPermissionResult.allowed && canEdit);
+
+  const tooltipText = !(editPermissionResult.allowed && canEdit)
+    ? 'Unauthorized to edit'
+    : '';
```

Comment:
@PatAKnight just a suggestion since this `!(editPermissionResult.allowed && canEdit)` is being used multiple times in this files so instead of the whole expression we can store the result in a const and use it wherever it is needed or you can use `disable` const as well.

```suggestion
 const disable = !(editPermissionResult.allowed && canEdit);
  const dataTestIdText = disable
    ? `disable-update-role-${roleName}`
    : `update-role-${roleName}`;
  const tooltipText = disable
    ? 'Unauthorized to edit'
    : '';
```